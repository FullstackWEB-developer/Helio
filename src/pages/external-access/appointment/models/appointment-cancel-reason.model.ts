export interface AppointmentCancelReason {
    appointmentCancelReasonId: number;
    name: string;
    noShow: boolean;
    patientRescheduled: boolean;
    providerUnavailable: boolean;
    slotAvailable: boolean;
}
