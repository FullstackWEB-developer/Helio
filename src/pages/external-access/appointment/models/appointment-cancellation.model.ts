export interface AppointmentCancellationModel {
    patientId: number;
    appointmentCancelReasonId: number;
    cancellationReason: string;
}
