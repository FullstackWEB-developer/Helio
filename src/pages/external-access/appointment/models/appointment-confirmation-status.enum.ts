export enum ConfirmationStatus {
    ConfirmedViaSMS = 161,
    ConfirmedViaEmail = 121
}