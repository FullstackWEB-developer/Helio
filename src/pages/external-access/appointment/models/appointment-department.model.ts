export interface AppointmentDepartmentModel {
    latLong: number[];
    name: string;
    address: string;
    address2?: string;
    city: string;
    state: string;
    zip: string;
}
