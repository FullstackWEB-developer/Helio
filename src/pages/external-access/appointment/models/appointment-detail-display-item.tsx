export interface AppointmentDetailDisplayItem {
    title: string,
    description: string
}
