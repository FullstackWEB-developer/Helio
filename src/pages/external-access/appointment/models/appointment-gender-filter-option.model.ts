
export enum GenderFilterOption {
    Female = 'GENDER_FEMALE',
    Male = 'GENDER_MALE'
}
