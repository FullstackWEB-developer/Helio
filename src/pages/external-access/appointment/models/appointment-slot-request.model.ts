export enum AppointmentSlotTimeOfDay {
    EarlyMorning = 1,
    Morning = 2,
    Afternoon = 3
}
export interface AppointmentSlotRequest {
    departmentId?: number;
    startDate: Date;
    endDate: Date;
    providerId?: number[];
    ignoreSchedulablePermission?: boolean;
    appointmentTypeId: number;
    itemCount?: number;
    patientId?: number;
    allowMultipleDepartment?: boolean;
    patientDefaultDepartmentId?: number;
    patientDefaultProviderId?: number
    timeOfDays?: AppointmentSlotTimeOfDay[];
    firstAvailable: boolean;
}

