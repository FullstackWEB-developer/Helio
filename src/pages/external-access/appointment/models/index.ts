export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
