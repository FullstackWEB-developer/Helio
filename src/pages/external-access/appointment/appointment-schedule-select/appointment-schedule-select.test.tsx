import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {act, fireEvent, render, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import AppointmentScheduleSelect from './appointment-schedule-select';
import Router from "react-router";
import Api from '@shared/services/api';
import { VerifiedPatient } from '@pages/patients/models/verified-patient';
import { Location, Provider } from '@shared/models';
import { Appointment, AppointmentSlotTimeOfDay, AppointmentType } from '../models';
import { PatientAppointmentType } from '@shared/models/patient-appointment-type.enum';
import { AppointmentDepartmentModel } from '../models/appointment-department.model';
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
  jest.setTimeout(70000);
describe("AppointmentScheduleSelect tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {
            providerList: [
                {
                    ansiNameCode: '',
                    ansiSpecialtyCode: '',
                    billable: false,
                    createEncounterOnCheckin: false,
                    firstName: 'provider',
                    lastName: 'lastname',
                    id: 3
                }
            ] as Provider[],
            locationList: [{
                medicationHistoryConsent: true,
                timeZoneOffset: 1,
                isHospitalDepartment: true,
                state: "string",
                portalUrl: "string",
                city: "string",
                placeOfServiceFacility: true,
                serviceDepartment: true,
                latitude: 1,
                fax: "string",
                doesNotObservedSt: true,
                parkingInformation: "string",
                id: 1,
                address: "string",
                placeOfServiceTypeId: 1,
                longitude: 1,
                clinicals: "string",
                timeZone: 1,
                patientDepartmentName: "string",
                chartSharingGroupId: "string",
                name: "string",
                placeOfServiceTypeName: "string",
                phone: "string",
                address2: "string",
                clinicalProviderFax: "string",
                ecommerceCreditCardTypes: ["1", "2"],
                zip: "string",
                timeZoneName: "string"
            }
            ] as Location[]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        },
        patientsState: {
            verifiedPatient: {
                firstName: "string",
                lastName: "string",
                patientId: 123,
                primaryProviderId: 12,
                departmentId: 1,
                defaultProviderId: 12,
                defaultDepartmentId: 1,
                dateOfBirth: new Date("1995-11-15"),
                token: "string",
                tokenExpiration: new Date("2023-11-15")
            } as VerifiedPatient
        },
        externalAccessState: {
            appointmentsState: {
                patientUpcomingAppointment: {
                    appointmentId: "string",
                    appointmentStatus: "Future",
                    appointmentType: "string",
                    appointmentTypeId:1,
                    copay:1,
                    date: new Date("2022-11-15"),
                    departmentId: 1,
                    duration:1,
                    patientAppointmentTypeName: "string",
                    providerId:1,
                    startTime: "string",
                    startDateTime: new Date("2022-11-15"),
                } as Appointment,
                appointmentSlotRequest: {
                    departmentId: 1,
                    startDate: new Date('2022-11-15'),
                    endDate: new Date('2022-11-15'),
                    providerId: [1],
                    ignoreSchedulablePermission: false,
                    appointmentTypeId: 1,
                    itemCount: 3,
                    patientId: 1,
                    allowMultipleDepartment: true,
                    patientDefaultDepartmentId: 10,
                    patientDefaultProviderId: 10,
                    timeOfDays: [AppointmentSlotTimeOfDay.EarlyMorning, AppointmentSlotTimeOfDay.Afternoon] as AppointmentSlotTimeOfDay[],
                    firstAvailable: true
                }
            }
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders AppointmentScheduleSelect correctly", async () => {
        let fragment: any;
        jest.spyOn(Api, 'get').mockImplementation((url) => {
            if (url.includes('open-slots')) {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            address1: 'address',
                            city: 'city',
                            phoneNumber: '2222222222',
                            state: 'AL',
                            zip: '55555',
                            clinicalProviderId: 3,
                            clinicalProviderName: 'bb',
                            faxNumber: '223232323',
                            departmentId: 1
                        }] as any[]
                    })
                })
            } else {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            stateCode: 'AL',
                            name: 'Alabama'
                        }]
                    })
                })
            }
        });
        await act(async () => {
            const {asFragment, getByTestId} = render(<TestWrapper mockState={mockState}>
                <AppointmentScheduleSelect/>          
            </TestWrapper>);
            fragment = asFragment;
            await waitFor(() => {
                fireEvent.click(getByTestId('schedule-select-filter'));
            });
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders AppointmentScheduleSelect correctly not empty slot", async () => {
        let fragment: any;
        jest.spyOn(Api, 'get').mockImplementation((url) => {
            if (url.includes('open-slots')) {
                return new Promise((resolve) => {
                    resolve({
                        data: [] as any[]
                    })
                })
            } else {
                return new Promise((resolve) => {
                    resolve({
                        data: [{
                            stateCode: 'AL',
                            name: 'Alabama'
                        }]
                    })
                })
            }
        });
        await act(async () => {
            const {asFragment, getByTestId} = render(<TestWrapper mockState={mockState}>
                <AppointmentScheduleSelect/>          
            </TestWrapper>);
            fragment = asFragment;
            await waitFor(() => {
                fireEvent.click(getByTestId('schedule-select-filter'));
            });
        });
        expect(fragment()).toMatchSnapshot();
    });
})
