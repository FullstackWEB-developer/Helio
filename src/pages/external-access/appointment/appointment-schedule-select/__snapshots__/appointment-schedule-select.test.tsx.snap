// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentScheduleSelect tests renders AppointmentScheduleSelect correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row appointment-schedule-select without-default-padding without-default-padding-right without-default-padding-bottom"
  >
    <div
      class="flex-1 pt-16 mr-14 pb-36"
    >
      <div
        class="flex items-center 2xl:whitespace-pre 2xl:h-12 2xl:mt-3 2xl:mb-9"
      >
        <h4>
          external_access.appointments.select_your_appointment
        </h4>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row flex-wrap items-center mb-4 lg:flex-nowrap"
        >
          <div
            class="date-time-picker w-full flex flex-col h-20 lg:mr-8 relative"
          >
            <div
              class="date-time-picker-container flex flex-wrap relative"
            >
              <input
                autocomplete="off"
                class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 activated cursor-pointer"
                data-testid="selectedDate"
                max="9999-12-31"
                min="2022-10-21"
                name="selectedDate"
                readonly=""
                type="text"
                value="Invalid Date"
              />
              <label
                class="absolute truncate"
              >
                <span
                  class="label-span body3"
                >
                  external_access.appointments.date
                </span>
              </label>
              <div
                class="input-addon px-3 pt-4"
                role="button"
              >
                <div
                  class=""
                >
                  <svg
                    class="icon-medium "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="date-time-picker-arrow"
                      d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                      transform="translate(5 16.273) rotate(-90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="select-wrapper relative w-full flex flex-col h-20  lg:mr-8"
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="on"
                autocorrect="on"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                data-testid="provider"
                name="provider"
                spellcheck="true"
                type="text"
                value=""
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="select-label-span body2"
                >
                  external_access.schedule_appointment.provider
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-common.any"
                      data-testid="select-cell-text-common.any"
                    >
                      common.any
                    </div>
                  </div>
                </div>
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-provider lastname"
                      data-testid="select-cell-text-provider lastname"
                    >
                      provider lastname
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="select-wrapper relative w-full flex flex-col h-20 "
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="on"
                autocorrect="on"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger activated  "
                data-testid="location"
                name="location"
                spellcheck="true"
                type="text"
                value="String"
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="select-label-span body3"
                >
                  external_access.schedule_appointment.location
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-common.any"
                      data-testid="select-cell-text-common.any"
                    >
                      common.any
                    </div>
                  </div>
                </div>
                <div
                  class="w-full select-cell justify-between flex items-center  body2-white   is-selected  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-String"
                      data-testid="select-cell-text-String"
                    >
                      String
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col lg:ml-7 items-center"
          >
            <div
              class="body2-medium"
            >
              external_access.schedule_appointment.filters.title
            </div>
            <div
              class="relative flex flex-row items-center"
            >
              <div
                class="pb-6 cursor-pointer flex justify-center"
              >
                <svg
                  class="icon-medium "
                  data-testid="schedule-select-filter"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <path
                    class="rgba-05-fill"
                    d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex-row hidden lg:flex"
          >
            <div
              class=""
            >
              <svg
                class="cursor-pointer "
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="active-item-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(8 5)"
                />
              </svg>
            </div>
            <div>
              <div
                class="grid grid-flow-col pb-3 auto-cols-max md:auto-cols-min"
              />
              <div
                class="grid justify-between w-full grid-flow-col pt-4 overflow-y-auto border-t auto-cols-max md:auto-cols-min slot-full-container"
              />
              <div
                class="flex justify-center w-full"
              />
            </div>
            <div
              class=""
            >
              <svg
                class="cursor-pointer "
                data-testid="svg-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="active-item-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(16.272 19) rotate(180)"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex flex-row lg:hidden"
          >
            <div
              class=""
            >
              <svg
                class="mt-1 cursor-pointer "
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="active-item-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(8 5)"
                />
              </svg>
            </div>
            <div
              class="w-full"
            >
              <div
                class="p-1"
              >
                <div
                  class="flex justify-center w-auto px-4 pb-3 mb-4 border-b subtitle2"
                >
                  Invalid Date
                </div>
                <div
                  data-test-id="external-access-appointments-no-day-slots"
                >
                  external_access.appointments.no_day_slots_found
                </div>
              </div>
            </div>
            <div
              class=""
            >
              <svg
                class="mt-1 cursor-pointer "
                data-testid="svg-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="active-item-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(16.272 19) rotate(180)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-center"
    >
      <div
        class="flex items-center justify-center w-full h-full"
      >
        <div />
      </div>
    </div>
    <div
      class="flex-1 hidden h-full lg:block"
    >
      <div
        class="map-viewer-wrapper"
      >
        <div
          class="map-viewer leaflet-container leaflet-grab leaflet-touch-drag"
          style="position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 1;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://a.tile.openstreetmap.org/18/131800/130343.png"
                    style="width: 256px; height: 256px; left: -46px; top: -201px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-shadow-pane"
            >
              <img
                alt=""
                class="leaflet-marker-shadow leaflet-zoom-hide"
                src="marker-shadow.png"
                style="left: 0px; top: 0px;"
              />
            </div>
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            >
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="marker-icon.png"
                style="left: 0px; top: 0px; z-index: 0;"
                tabindex="0"
              />
            </div>
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            >
              <div
                class="leaflet-control-zoom leaflet-bar leaflet-control"
              >
                <a
                  aria-label="Zoom in"
                  class="leaflet-control-zoom-in leaflet-disabled"
                  href="#"
                  role="button"
                  title="Zoom in"
                >
                  +
                </a>
                <a
                  aria-label="Zoom out"
                  class="leaflet-control-zoom-out"
                  href="#"
                  role="button"
                  title="Zoom out"
                >
                  −
                </a>
              </div>
            </div>
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JS library for interactive maps"
                >
                  Leaflet
                </a>
                 | © 
                <a
                  href="http://osm.org/copyright"
                >
                  OpenStreetMap
                </a>
                 contributors
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduleSelect tests renders AppointmentScheduleSelect correctly not empty slot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row appointment-schedule-select without-default-padding without-default-padding-right without-default-padding-bottom"
  >
    <div
      class="flex-1 pt-16 mr-14 pb-36"
    >
      <div
        class="flex items-center 2xl:whitespace-pre 2xl:h-12 2xl:mt-3 2xl:mb-9"
      >
        <h4>
          external_access.appointments.select_your_appointment
        </h4>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-row flex-wrap items-center mb-4 lg:flex-nowrap"
        >
          <div
            class="date-time-picker w-full flex flex-col h-20 lg:mr-8 relative"
          >
            <div
              class="date-time-picker-container flex flex-wrap relative"
            >
              <input
                autocomplete="off"
                class="relative truncate flex-shrink flex-grow flex-auto leading-normal w-px pl-4 body2 input-pt h-14 activated cursor-pointer"
                data-testid="selectedDate"
                max="9999-12-31"
                min="2022-10-21"
                name="selectedDate"
                readonly=""
                type="text"
                value="Mon, Jan 9 2023"
              />
              <label
                class="absolute truncate"
              >
                <span
                  class="label-span body3"
                >
                  external_access.appointments.date
                </span>
              </label>
              <div
                class="input-addon px-3 pt-4"
                role="button"
              >
                <div
                  class=""
                >
                  <svg
                    class="icon-medium "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="date-time-picker-arrow"
                      d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                      transform="translate(5 16.273) rotate(-90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="select-wrapper relative w-full flex flex-col h-20  lg:mr-8"
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="on"
                autocorrect="on"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger   "
                data-testid="provider"
                name="provider"
                spellcheck="true"
                type="text"
                value=""
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="select-label-span body2"
                >
                  external_access.schedule_appointment.provider
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-common.any"
                      data-testid="select-cell-text-common.any"
                    >
                      common.any
                    </div>
                  </div>
                </div>
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-provider lastname"
                      data-testid="select-cell-text-provider lastname"
                    >
                      provider lastname
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="select-wrapper relative w-full flex flex-col h-20 "
          >
            <div
              class="select relative flex flex-col "
            >
              <input
                autocomplete="on"
                autocorrect="on"
                class="pl-4 pr-8 body2 select-trigger-pt h-14 relative truncate select-trigger activated  "
                data-testid="location"
                name="location"
                spellcheck="true"
                type="text"
                value="String"
              />
              <label
                class="select-label absolute truncate"
              >
                <span
                  class="select-label-span body3"
                >
                  external_access.schedule_appointment.location
                </span>
              </label>
              <div
                class="absolute pt-4 right-4 "
              >
                <div
                  class="flex flex-row items-start"
                >
                  <div
                    class="cursor-pointer"
                  >
                    <div
                      class=""
                    >
                      <svg
                        class="icon-medium "
                        height="24"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="none"
                          height="24"
                          width="24"
                        />
                        <path
                          class="select-arrow-fill"
                          d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                          transform="translate(5 16.273) rotate(-90)"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="absolute py-2 options"
              >
                <div
                  class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-common.any"
                      data-testid="select-cell-text-common.any"
                    >
                      common.any
                    </div>
                  </div>
                </div>
                <div
                  class="w-full select-cell justify-between flex items-center  body2-white   is-selected  "
                >
                  <div
                    class="flex flex-col justify-center w-full"
                    data-test-id="select-cell-icon-content"
                  >
                    <div
                      class="flex items-center pl-4"
                      data-test-id="select-cell-text-String"
                      data-testid="select-cell-text-String"
                    >
                      String
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col lg:ml-7 items-center"
          >
            <div
              class="body2-medium"
            >
              external_access.schedule_appointment.filters.title
            </div>
            <div
              class="relative flex flex-row items-center"
            >
              <div
                class="pb-6 cursor-pointer flex justify-center"
              >
                <svg
                  class="icon-medium "
                  data-testid="schedule-select-filter"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <path
                    class="rgba-05-fill"
                    d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div>
            <div>
              external_access.appointments.no_appointment_found_in_max_tries
            </div>
            <div>
              external_access.appointments.no_appointment_found_in_max_tries_2
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-center"
    >
      <div
        class="flex items-center justify-center w-full h-full"
      >
        <div />
      </div>
    </div>
    <div
      class="flex-1 hidden h-full lg:block"
    >
      <div
        class="map-viewer-wrapper"
      >
        <div
          class="map-viewer leaflet-container leaflet-grab leaflet-touch-drag"
          style="position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            />
            <div
              class="leaflet-pane leaflet-shadow-pane"
            />
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            />
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            >
              <div
                class="leaflet-control-zoom leaflet-bar leaflet-control"
              >
                <a
                  aria-label="Zoom in"
                  class="leaflet-control-zoom-in"
                  href="#"
                  role="button"
                  title="Zoom in"
                >
                  +
                </a>
                <a
                  aria-label="Zoom out"
                  class="leaflet-control-zoom-out"
                  href="#"
                  role="button"
                  title="Zoom out"
                >
                  −
                </a>
              </div>
            </div>
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JS library for interactive maps"
                >
                  Leaflet
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
